### ISSUE TITLE
Provided test set output is different from output from the pre-trained models. 

### ISSUE THREAD

**Issue by LeenaShekhar** [abisee/pointer-generator#63]
1. I used the pre-trained models (TF 1.2.1) to decode and calculated ROUGE scores for those. These outputs look different from the ones provided under the test_output folder. 

For example:

REFERENCE SUMMARY: in an interview with the new york times , president obama says he understands israel feels particularly vulnerable . obama calls the nuclear deal with iran a `` once-in-a-lifetime opportunity '' israeli prime minister benjamin netanyahu and many u.s. republicans warn that iran can not be trusted .

**Generated summary using the pre-trained model for pre-coverage:**

president barack obama says he is `` absolutely committed to making sure '' israel maintains a military advantage over iran . obama says he is `` absolutely committed to making sure they maintain their qualitative military edge ''

**Generated summary under test_output/pre-coverage folder:**

president barack obama says he is `` absolutely committed to making sure '' israel maintains a military advantage over iran .
obama said he understands and respects netanyahu 's stance that israel is particularly vulnerable and does n't `` have the luxury of testing these propositions '' .

2. Because of the difference in generated summaries ROUGE scores look different as well. 

**Reported ROUGE scores (calculated on the given output under test_output/pre-coverage folder):**

rouge_1_f_score: 0.3644 with confidence interval (0.3619, 0.3666)
rouge_1_recall: 0.3760 with confidence interval (0.3732, 0.3787)
rouge_1_precision: 0.3776 with confidence interval (0.3749, 0.3804)

ROUGE-2:
rouge_2_f_score: 0.1566 with confidence interval (0.1543, 0.1589)
rouge_2_recall: 0.1612 with confidence interval (0.1587, 0.1636)
rouge_2_precision: 0.1631 with confidence interval (0.1607, 0.1655)

ROUGE-l:
rouge_l_f_score: 0.3342 with confidence interval (0.3319, 0.3366)
rouge_l_recall: 0.3446 with confidence interval (0.3418, 0.3473)
rouge_l_precision: 0.3466 with confidence interval (0.3439, 0.3493)

**ROUGE scores on the output from the pre-trained model**

ROUGE-1:
rouge_1_f_score: 0.2660 with confidence interval (0.2584, 0.2738)
rouge_1_recall: 0.3211 with confidence interval (0.3114, 0.3307)
rouge_1_precision: 0.2367 with confidence interval (0.2296, 0.2440)

ROUGE-2:
rouge_2_f_score: 0.0946 with confidence interval (0.0881, 0.1012)
rouge_2_recall: 0.1153 with confidence interval (0.1073, 0.1239)
rouge_2_precision: 0.0834 with confidence interval (0.0777, 0.0893)

ROUGE-l:
rouge_l_f_score: 0.2368 with confidence interval (0.2295, 0.2443)
rouge_l_recall: 0.2862 with confidence interval (0.2769, 0.2955)
rouge_l_precision: 0.2105 with confidence interval (0.2036, 0.2177)

**Comment by WasifaChowdhury** [comment]
I used the pretrained model directly without further training for decoding as mentioned in README file,

python pointer-generator/run_summarization.py --mode=decode --data_path=finished_files_pointer-generator/chunked/test_* --vocab_path=finished_files_pointer-generator/vocab --log_root=experiments/pretrained_model_tf1.2.1 --exp_name=pretrained_model_tf1.2.1 --max_enc_steps=400 --max_dec_steps=120 --coverage=1 --single_pass=1

But the generated outputs after running don't match the ones given in the pretrained folder:

REFERENCE SUMMARY:
in an interview with the new york times , president obama says he understands israel feels particularly vulnerable . obama calls the nuclear deal with iran a `` once-in-a-lifetime opportunity '' israeli prime minister benjamin netanyahu and many u.s. republicans warn that iran can not be trusted .

GENERATED SUMMARY is: 
washington -lrb- cnn -rrb- president obama to to to to to the the the .
to the the the the the the the the the [UNK] .
the the the the the the [UNK] [UNK] [UNK] [UNK] [UNK] [UNK] .

Despite loading checkpoints from model-238410 (after coverage model), @LeenaShekhar my outputs are worse than yours, so was wondering whether you trained the model further or just used the pretrained model directly. Any help regarding this would be helpful as I need to reproduce outputs same as those already in the folder. Thanks.

**Comment by LeenaShekhar** [comment]
In my case generated summaries were different but not bad like yours. I have a feeling you are not loading the ckpt, check your ckpt path and make sure graph is initialized with those values rather than random ones.

**Comment by WasifaChowdhury** [comment]
I did run again and  check the ckpt path, it prints

INFO:tensorflow:Loading checkpoint experiments/pretrained_model_tf1.2.1/pretrained_model_tf1.2.1/train/model-238410
INFO:tensorflow:Restoring parameters from experiments/pretrained_model_tf1.2.1/pretrained_model_tf1.2.1/train/model-238410

So the path it uses is for model-238410, which is the final ckpt after coverage. The folder is named, decode_test_400maxenc_4beam_35mindec_120maxdec_ckpt-238410, which means it uses ckpt-238410. But the generated summaries are bad as before. I can't figure out the problem, I didn't change anything in the code, when running. I specify the path of ckpt in pretrained model, so 

python pointer-generator/run_summarization.py --mode=decode --data_path=finished_files_pointer-generator/chunked/test_* --vocab_path=finished_files_pointer-generator/vocab --log_root=experiments/pretrained_model_tf1.2.1 --exp_name=pretrained_model_tf1.2.1 --max_enc_steps=400 --max_dec_steps=120 --coverage=1 --single_pass=1

Did you change the beam_size from 4 to higher value? Or any changes you made when running? @LeenaShekhar  @abisee Thanks.

**Comment by huyingxi** [comment]
I have the same question, using the pre-trained model provided on the website to make predictions, and the results obtained are quite different from the results in the paper.
I don't know if my operation is incorrect or for other reasons.

**Comment by LeenaShekhar** [comment]
Hi @huyingxi, I am not sure how different is "quite different" for you. I am not working on it currently, but as you could see from my post above things were different by 1.5% in terms of results when I simply ran the pre-trained model without introducing any changes from my end. 

I have a feeling since code base was re-written due to TF version switch, model might have been affected. This was with the model 5-6 months back though. 

All the best.

**Comment by hanghang2333** [comment]
I guess it may be that the results of the pretrained model using the Python2 version of TensorFlow training were different when invoked by Python3, but I'm not sure. Hope someone can verify.（In fact, I have actually encountered this situation before.）

**Comment by LeenaShekhar** [comment]
Sorry I did not mean Python2 and Python3 difference. I think the author ported the code from TF older version to new version and that might have introduced something. Again, this is my guess  based on a few mail exchange with the author long back on a different issue.

**Comment by hanghang2333** [comment]
@LeenaShekhar hello,I have another problem.When testing Rouge scores, ground truth has a few sentences, that is, multiple @highlight separated in the original data. Looking at the code during training is to combine multiple ground truth sentences into one for training. So when testing Rouge scores, should I put multiple sentences in ground truth like [some_name.A.001.txt
some_name.B.001.txt
In some files of some_name.C.001.txt].Or combine them into a single file. And as for the results of the tests which I put them into a single file, the Rouge1 score and the Rouge2 score are still normal, but the Rouge L score is very low.
